lib/ansible/modules/cloud/amazon/elb_application_lb_facts.py

```diff
@@ -0,0 +1,260 @@
+        state:
+            description: The state of the load balancer.
+            returned: when status is present
+            type: dict
+            sample: "{'code': 'active'}"
+        tags:
+            description: The tags attached to the load balancer.
+            returned: when status is present
+            type: dict
+            sample: "{
+                'Tag': 'Example'
+            }"
+        type:
+            description: The type of load balancer.
+            returned: when status is present
+            type: string
+            sample: application
+        vpc_id:
+            description: The ID of the VPC for the load balancer.
+            returned: when status is present
+            type: string
+            sample: vpc-0011223344
+'''
+
+import traceback
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (boto3_conn, boto3_tag_list_to_ansible_dict, camel_dict_to_snake_dict,
+                                      ec2_argument_spec, get_aws_connection_info)
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, NoCredentialsError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def get_load_balancer_attributes(connection, module, load_balancer_arn):
+
+    try:
+        load_balancer_attributes = boto3_tag_list_to_ansible_dict(connection.describe_load_balancer_attributes(LoadBalancerArn=load_balancer_arn)['Attributes'])
+    except ClientError as e:
+        module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+    # Replace '.' with '_' in attribute key names to make it more Ansibley
+    for k, v in load_balancer_attributes.items():
+        load_balancer_attributes[k.replace('.', '_')] = v
+        del load_balancer_attributes[k]
+
+    return load_balancer_attributes
+
+
+def get_load_balancer_tags(connection, module, load_balancer_arn):
+
+    try:
+        return boto3_tag_list_to_ansible_dict(connection.describe_tags(ResourceArns=[load_balancer_arn])['TagDescriptions'][0]['Tags'])
+    except ClientError as e:
+        module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+
+def list_load_balancers(connection, module):
+
+    load_balancer_arns = module.params.get("load_balancer_arns")
+    names = module.params.get("names")
+
+    try:
+        load_balancer_paginator = connection.get_paginator('describe_load_balancers')
+        if not load_balancer_arns and not names:
+            load_balancers = load_balancer_paginator.paginate().build_full_result()
+        if load_balancer_arns:
+            load_balancers = load_balancer_paginator.paginate(LoadBalancerArns=load_balancer_arns).build_full_result()
+        if names:
+            load_balancers = load_balancer_paginator.paginate(Names=names).build_full_result()
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'LoadBalancerNotFound':
```

Comment:
True.

I don't want to query for each ELB separately though.  I can't think of any other option though.